### DISCUSSION TITLE
CreateConnection hangs under Orleans Runtime

### DISCUSSION THREAD

**Discussion by snovak7** [rabbitmq/rabbitmq-dotnet-client#1343] [General]
### Describe the bug

When writing an adapter for Orleans Framework, I stumbled upon very specific TimeoutException which is hardcoded to 60minutes, this only happens inside Orleans contained runtime.

Having a more simplistic integration test which doesn't use Orleans, it worked without problems, as described in the documentation.

For now I downgraded to version 5.2.0 where it is working under such conditions.

### Reproduction steps

1. Pull Solution in development from https://github.com/snovak7/Escendit.Orleans.Streaming.RabbitMQ
2. Update the NuGet Dependencies to latest version (6.x) currently set with version of 5.2.0
3. Run one of OrleansTests integration tests which uses stream provider (needs local RabbitMQ instance)
...

### Expected behavior

CreateConnection returns the connection instead of waiting time

### Additional context

It occurred on my two Windows 11 machines
Orleans 7.1.2 is used with .NET 7
I used the latest RabbitMQ Server 3.11.13 on Windows
I also implemented the Stream Client version, which works perfectly.

**Comment by lukebakken** [comment]
@snovak7 thanks for using RabbitMQ and this library. As of yet there is no evidence that this behavior is due to an issue with this library, so I have converted the issue to a discussion.

> I also implemented the Stream Client version, which works perfectly

I'm not sure what this means. Does this mean you used [this library](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client) in your framework successfully?

> Run one of OrleansTests integration tests which uses stream provider

Can you please tell me _exactly_ how to do this so I don't spend time trying to figure out how? Thanks.

**Reply by snovak7** [reply]
> @snovak7 thanks for using RabbitMQ and this library. As of yet there is no evidence that this behavior is due to an issue with this library, so I have converted the issue to a discussion.
> 
> > I also implemented the Stream Client version, which works perfectly
> 
> I'm not sure what this means. Does this mean you used [this library](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client) in your framework successfully?
>

That's right, I used the other library to have 2 options when using stream provider.

> > Run one of OrleansTests integration tests which uses stream provider
> 
> Can you please tell me _exactly_ how to do this so I don't spend time trying to figure out how? Thanks.

There are 5 tests only as of now,
These 3 connect to RabbitMQ: ClientDirectStreamAsync, ClientStreamTestAsync, SiloStreamTestAsync

Hope this helps!

**Reply by lukebakken** [reply]
Thanks, we'll investigate when time allows.

**Reply by lukebakken** [reply]
Yep if I run `ClientDirectStreamAsync` I can see the connection is established to RabbitMQ, but the AMQP handshake never completes. No idea why.

**Reply by snovak7** [reply]
I have wrapped the call to use Default TaskScheduler, then it works.

**Reply by lukebakken** [reply]
No kidding. I would never have figured that out. Thanks.
